wagl/ancillary.py

```diff
@@ -696,20 +697,41 @@ def get_water_vapour(acquisition, water_vapour_dict, scale_factor=0.1):
     if band == 0 and doy == 1:
         band = 1
 
-    # Get the number of bands
-    with rasterio.open(datafile) as src:
-        n_bands = src.count
+    try:
+        # Get the number of bands
+        with rasterio.open(datafile) as src:
+            n_bands = src.count
 
-    # Enable NBAR Near Real Time (NRT) processing
-    if band > (n_bands + 1):
-        rasterdoy = (((n_bands) - (int((hour + 3) / 6))) / 4) + 1
-        if (doy - rasterdoy) < 7:
-            band = (int(rasterdoy) - 1) * 4 + int((hour + 3) / 6)
+        # Enable NBAR Near Real Time (NRT) processing (7 day window)
+        if band > (n_bands + 1):
+            rasterdoy = (((n_bands) - (int((hour + 3) / 6))) / 4) + 1
+            if (doy - rasterdoy) < 7:
+                band = (int(rasterdoy) - 1) * 4 + int((hour + 3) / 6)
 
-    try:
         data = get_pixel(datafile, geobox.centre_lonlat, band=band)
-    except IndexError:
-        raise AncillaryError("No Water Vapour data")
+    except (IndexError, rasterio.errors.RasterioIOError):
+        # Fallback for NRT or whenever we have no real data
+        # key name 'fallback_data' can change. Quickest name i could think of
+        if 'fallback_data' not in water_vapour_dict:
+            raise AncillaryError("No actual or fallback water vapour data.")
+        else:
+            # maybe a seperate func, but here will do for the time being
+            month = dt.strftime('%B-%d').upper()
+
+            # closest previous observation
+            # i.e. observations are at 0000, 0600, 1200, 1800
+            # and an acquisition hour of 1700 will use the 1200 observation
+            observations = numpy.array([0, 6, 12, 18])
+            hr = observations[numpy.argmin(numpy.abs(hour - observations))]
+            dname = 'AVERAGE/{}/{:02d}00'.format(month, hr)
+
+            with h5py.File(water_vapour_dict['fallback_data'], 'r') as fid:
+                ds = fid[dname]
+                transform = Affine.from_gdal(*ds.attrs['geotransform'])
+                x, y = [int(v) for v in ~transform * geobox.centre_lonlat]
+                data = ds[y, x]
+                url = urlparse(water_vapour_dict['fallback_data'],
+                               scheme='file').geturl()
 
```

Comment:
need to update the reference for the datafile in order to add the required header information.

The reference to the datafile will also need to be updated to the fallback file in order to capture the associated metadata.

```suggestion
                datafile = water_vapour_dict['fallback_data']

```

Reply:
Not sure about the alignment of this suggested edit